query_executor.go

```diff
@@ -122,16 +123,17 @@ func (q *queryExecutor) executeQuery(qry ExecutableQuery) (*Iter, error) {
 	case iter := <-results:
 		return iter, nil
 	case <-ctx.Done():
-		return &Iter{err: ctx.Err()}, nil
+		return NewIterErr(ctx.Err()), nil
 	}
 }
 
-func (q *queryExecutor) do(ctx context.Context, qry ExecutableQuery, hostIter NextHost) *Iter {
+func (q *queryExecutor) do(ctx context.Context, qry ExecutableQuery, it *Iter, hostIter NextHost) *Iter {
```

Comment:
Since we move a lot of "mutable" query execution part into `Iter`, I think it does make sense to add it to the method chain that executes given query.